### PR TITLE
update cockroach-cloud-sdk-go to v4

### PR DESCRIPTION
**Pull request by dcrosta**
Imported latest OpenAPI definitions and regenerated code. See CHANGELOG.md for details on what has changed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,13 +13,13 @@ For more information, please visit [https://support.cockroachlabs.com](https://s
 Install the package:
```
Comment (dcrosta) [inline README.md:13]: @fantapop should L6/L7 be updated here? (Github won't let me comment on those lines as they're outside the diff...)

Comment (fantapop) [reply README.md:13]: hmm, yeah I wonder why that's not updated.  That file is generated from this template.  Here is the line:

https://github.com/cockroachdb/cockroach-cloud-sdk-go/blob/main/internal/spec/templates/README.mustache#L7

Comment (fantapop) [reply README.md:13]: It looks like this needs to be updated: https://github.com/cockroachdb/cockroach-cloud-sdk-go/blob/e2fef1f9062095e3e71531634ea65145e3943557/internal/spec/config.yaml#L2

File: internal/openapi-generator/api/openapi.yaml [hunk 2]
```diff
@@ -9418,9 +9420,8 @@ components:
         in the type URL, for example "foo.bar.com/x/y.z" will yield type
         name "y.z".
 
-
         JSON
-
+        ====
```
Comment (fantapop) [inline internal/openapi-generator/api/openapi.yaml:9424]: Hmm, this is surprising.  There wasn't a merge conflict here was there?

**Review by fantapop — APPROVED** [review]
These look good.  I did have one question about the ==== that was added as part of the generation.

File: internal/openapi-generator/api/openapi.yaml (hunk 2 above)
Comment (dcrosta) [reply internal/openapi-generator/api/openapi.yaml:9424]: That is in and seems to come from openapi.json. That big blob of text seems to come from the protobuf description of `Any`, if I'm reading it right -- probably a change in protobuf library version introduced this small docs change? In any event, I don't think it's due to a merge conflict or anything like that.

**Pull request merged**